lib/utils.js

```diff
@@ -140,28 +145,35 @@ const utils = (module.exports = {
         );
 
         if (extraKeys.length) {
-          emitWarning(
-            `Invalid options found (${extraKeys.join(', ')}); ignoring.`
-          );
+          const nonDeprecated = extraKeys.filter((key) => {
+            if (!DEPRECATED_OPTIONS[key]) {
+              return true;
+            }
+            const newParam = DEPRECATED_OPTIONS[key];
+            /**
+             * TODO turn this into a hard error in a future major version (once we have fixed our docs).
+             */
+            emitWarning(`'${key}' is deprecated; use '${newParam}' instead.`);
+            params[newParam] = params[key];
```

Comment:
Should this hard error if params[key] is already set?

Reply:
Interesting... yeah I think that's sane / unlikely to cause much user frustration / generally more likely to be helpful than harmful.

Reply:
(Done)